packages/nodes-base/nodes/FunctionItem/FunctionItem.node.ts

```diff
@@ -75,22 +75,68 @@ return item;`,
 		};
 
 		for (let itemIndex = 0; itemIndex < length; itemIndex++) {
+			const mode = this.getMode();
+
 			try {
 				item = items[itemIndex];
+				item.index = itemIndex;
 
 				// Copy the items as they may get changed in the functions
 				item = JSON.parse(JSON.stringify(item));
 
 				// Define the global objects for the custom function
 				const sandbox = {
+					/** @deprecated for removal */
 					getBinaryData: (): IBinaryKeyData | undefined => {
+						if (mode === 'manual')
+							this.sendMessageToUI(
+								'getBinaryData(...) is deprecated and will be removed in a future version. Please consider switching to getBinaryDataAsync(...) instead.',
+							);
 						return item.binary;
 					},
+					/** @deprecated for removal */
+					setBinaryData: async (data: IBinaryKeyData) => {
+						if (mode === 'manual')
+							this.sendMessageToUI(
+								'setBinaryData(...) is deprecated and will be removed in a future version. Please consider switching to setBinaryDataAsync(...) instead.',
+							);
+						item.binary = data;
+					},
 					getNodeParameter: this.getNodeParameter,
 					getWorkflowStaticData: this.getWorkflowStaticData,
 					helpers: this.helpers,
 					item: item.json,
-					setBinaryData: (data: IBinaryKeyData) => {
+					getBinaryDataAsync: async (): Promise<IBinaryKeyData | undefined> => {
```

Comment:
> getBinaryDataAsync: async ()

The `Function` node requires one to pass in the item here. It isn't necessary in the `FunctionItem` node, however perhaps we should align the two for better a better user experience when switching between them? Though - to be honest - I quite like it this way if we think about the nodes independently.